design/virtualization.md

```diff
@@ -0,0 +1,133 @@
+for example `pc` and `q35` on x86 systems, `virt` on ARM systems and `pseries` on IBM Power systems. The default Kata Containers
+machine type is `pc`. The machine type and its [`Machine accelerators`](#machine-accelerators) can
+be changed by editing the runtime [`configuration`](#configuration) file.
+
+#### Devices and features used
+
+- virtio-vsock or virtio-serial
+- virtio-blk or virtio-scsi
+- virtio-net
+- virtio-fs or virtio-9p (recommend: virtio-fs)
+- VFIO
+- hotplug
+- machine accelerators
+
+Machine accelerators and hotplug are used in Kata Containers to manage resource constraints, improve boot time and reduce memory footprint. These are documented below. 
+
+#### Machine accelerators
+
+Machine accelerators are architecture specific and can be used to improve the performance
+and enable specific features of the machine types. The following machine accelerators
+are used in Kata Containers:
+
+- NVDIMM: This machine accelerator is x86 specific and only supported by `pc` and
+`q35` machine types. `nvdimm` is used to provide the root filesystem as a persistent
+memory device to the Virtual Machine.
+
+#### Hotplug devices
+
+The Kata Containers VM starts with a minimum amount of resources, allowing for faster boot time and a reduction in memory footprint.  As the container launch progresses, devices are hotplugged to the VM. For example, when a CPU constraint is specified which includes additional CPUs, they can be hot added.  Kata Containers has support for hot-adding the following devices:
+- Virtio block
+- Virtio SCSI
+- VFIO
+- CPU
+
+### Firecracker/KVM
+
+Firecracker, built on many rust crates that are within [rust-vmm](https://github.com/rust-vmm),  has a very limited device model, providing a lighter
+footprint and attack surface, focusing on FaaS like use cases. As a result, Kata Containers with Firecracker VMM supports a subset of the CRI API.
+Firecracker does not support file-system sharing, and as a result only block-based storage drivers are supported. Firecracker does not support device
+hotplug nor does it support VFIO. As a result, Kata Containers with Firecracker VMM does not support updating container resources after boot, nor
+does it support device passthrough.
+
+#### Devices used
+
+- virtio-vsock
+- virtio-blk
+- virtio-net
+
+### Cloud Hypervisor/KVM
+
+Cloud Hypervisor, based on [rust-vmm](https://github.com/rust-vmm), is designed to have a lighter footprint and attack surface. For Kata Containers,
+relative to Firecracker, the Cloud Hypervisor configuration provides better compatability at the expense of exposing additional devices: file system
+sharing and direct device assignment.  As of the 1.10 release of Kata Conatiners, Cloud Hypervisor does not support device hotplug, and as a result
+does not support updating container resources after boot. See [Cloud Hypervisor device support documentation](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/docs/device_model.md)
+for more details.
```

Comment:
#### Limitations
##### Device hotplug
 CPU, memory, block devices have not support for hotplug, after Cloud Hypervisor enable this features will be enabled in Kata as well.
##### No block device support for volumes.
T he block device is not supported via hot plug, current support block device as with hot-plug does not allow cases like run in Kubernetes where each container rootfs volume could be hotpluged as a block device 
##### No guest serial logs via shim
Guest logs are not collected by shim today, this will be added in near future.

Reply:
Addressed (mostly).  Since we don't get into that level of detail with others, I'm happy to keep this lighter.